ClothesApp/Infrastructure/Repositories/BrandsRepository.cs

```diff
@@ -0,0 +1,64 @@
+using Application.Repositories;
+using Domain.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace Infrastructure.Repositories;
+
+public class BrandsRepository : IBrandsRepository
+{
+    private readonly ClothesAppContext _dbContext;
+
+    public BrandsRepository(ClothesAppContext context)
+    {
+        _dbContext = context;
+    }
+
+    public async Task<Brand> Insert(Brand brand)
+    {
+        await _dbContext.Brands.AddAsync(brand);
+        await _dbContext.SaveChangesAsync();
+
+        return brand;
+    }
+
+    public async Task<Brand> Update(Brand brand)
+    {
+        _dbContext.Update(brand);
+        await _dbContext.SaveChangesAsync();
+
+        return brand;
+    }
+    
+    public async Task<Brand> DeleteBrandById(long id)
+    {
+        var brand = await _dbContext.Brands.FindAsync(id);
+
+        if (brand is not null)
+        {
+            _dbContext.Brands.Remove(brand);
+            await _dbContext.SaveChangesAsync();
+        }
+
+        return brand;
+    }
+
+    public async Task<Brand> GetBrandById(long id)
+    {
+        return await _dbContext.Brands.AsNoTracking().FirstOrDefaultAsync(b => b.Id == id);
```

Comment:
AsNoTracking on a single get is very bad decision, because the default way to update entities is to get them, update their properties and then set back
If you want to just check if brands exists but don't track the entity then use DoesExist like you do in some places